tutorials/developer-docs/adding-models.md

```diff
@@ -0,0 +1,258 @@
+# Adding New Models
+
+This document provides an overview and explanation of how new LLM architectures and model weights can be added to LitGPT.
+
+&nbsp;
+
+> [!NOTE]
+> One of the design focus areas of LitGPT is to provide efficient readable code. At the same time, LitGPT aims to support selected LLMs that are useful to the community. LitGPT aims to reuse and share as much code as possible between different LLMs to strike a balance between code readability and enabling support for various LLMs. In short, we try to minimize writing custom code for a given LLM and aim for code reuse.
+
+
+&nbsp;
+
+&nbsp;
+## 1. Discuss the LLM to be added
+
+As an open-source project, we appreciate your contributions! However, before you begin putting valuable time and work into a contribution, ideally, open an issue to discuss whether support for a certain model is within the project's scope.
+
+&nbsp;
+## 2. Set up your development environment
+
+Clone the repository:
+
+```bash
+git clone https://github.com/Lightning-AI/litgpt.git
+```
+
+Then, install it with the "editable" mode for development:
+
+```bash
+cd litgpt
+pip install litgpt -e ".[all]"
+```
+
+&nbsp;
+## 3. Update the config file
+
+Update the [../litgpt/config.py](litgpt/config.py) config file, adding the new model configuration there. It's easiest to start with the most similar model, copy the configuration, and then modify it according to the `config.json` file on the HF hub.
+
+For example, suppose an entry for Llama 3 8B already exists and you want to add support for Llama 3 70B. 
+
+Copy the Llama 3 8B entry:
+
+```python
+ # https://huggingface.co/meta-llama/Meta-Llama-3-8B/blob/main/config.json
+ dict(
+     name="Llama-3-8B{}",
+     hf_config=dict(org="meta-llama", name="Meta-Llama-3-8B{}"),
+     vocab_size=128256,
+     padding_multiple=64,
+     n_layer=32,
+     n_head=32,
+     n_query_groups=8,
+     rotary_percentage=1.0,
+     parallel_residual=False,
+     bias=False,
+     norm_class_name="RMSNorm",
+     mlp_class_name="LLaMAMLP",
+     intermediate_size=14336,
+     rope_base=500000,
+ ),
+```
+
+Then create the entry for the 70B model. Here, make sure you update the values according to the `config.json` file available on the HF hub:
+
+```python
+# https://huggingface.co/meta-llama/Meta-Llama-3-70B/blob/main/config.json
+ dict(
+     name="Llama-3-70B{}",
+     hf_config=dict(org="meta-llama", name="Meta-Llama-3-70B{}"),
+     vocab_size=128256,
+     padding_multiple=64,
+     n_layer=80,
+     n_head=64,
+     n_embd=8192,
+     n_query_groups=8,
+     rotary_percentage=1.0,
+     parallel_residual=False,
+     bias=False,
+     norm_class_name="RMSNorm",
+     mlp_class_name="LLaMAMLP",
+     intermediate_size=28672,
+     rope_base=500000,
+ ),
+```
+
+&nbsp;
+
+> [!NOTE]
+> Some models may require you to implement a new MLP class analogous to `class LLaMAMLP`.
```

Comment:
A good indicator is existing modeling.py file.
If it exists (in the model's repo), than something in the current version of HF doesn't support the new model and that will require additional changes besides just updating the config.

Reply:
Good idea, I added this!